LPS-143179 Migrate FDS headless display in `objects-web`

markocikos opened the pull request:
JIRA: https://issues.liferay.com/browse/LPS-143179

@liferay-objects In this PR, we are migrating `<clay:headless-data-set-display>` to `<frontend-data-set:headless-display>`. As a part of the migration, we are changing backend API from `frontend-taglib-clay` to API from `frontend-data-set` app. There is **no change in logic**, just a switch of tag and API classes, along with rename of variables to match the new API.

This is partly a followup on https://github.com/liferay-frontend/liferay-portal/pull/1722, as we are porting to `frontend-data-set` changes to API introduced in LPS-142886.

This is one of three parts of the migration:
- [x]  `search-experiences`
- [x] `objects`
- [ ] `commerce`

I tested changes on JSPs I could find in UI:
- [x]  view_object_definitions.jsp
- [x]  fields.jsp
- [x]  actions.jsp
- [x]  picklists > view.jsp
- [x]  edit_list_type_definition.jsp
- [x] view_object_entries
- [x] relationships
- [x] layouts

Please check the rest!

/cc @LuismiBarcos @javierdearcos

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-28.liferay.com/job/test-portal-source-format/1111/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-28.liferay.com/userContent/jobs/test-portal-source-format/builds/1111/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 5 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/a3a615adb4df920316561cec06558676c7e94377">a3a615adb4df920316561cec06558676c7e94377</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/markocikos/liferay-portal/tree/LPS-143179-2">LPS-143179-2</a><br/>Branch GIT ID: <a href="https://github.com/markocikos/liferay-portal/commit/f8d811f449b5a66607fa10fa574d669206f89644">f8d811f449b5a66607fa10fa574d669206f89644</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-28.liferay.com/job/test-portal-source-format/1111/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-28.liferay.com/userContent/jobs/test-portal-source-format/builds/1111/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-28.liferay.com/job/test-portal-source-format/1111/">test-portal-source-format#1111</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-28.liferay.com/userContent/jobs/test-portal-source-format/builds/1111/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-objects/liferay-portal/pull/298">liferay-objects#298</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1592714335">[master] ci:test:sf - markocikos &gt; liferay-objects - PR#298 - 2021-12-20[09:08:47]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-28.liferay.com/job/publish-testray-report/3871/">publish-testray-report#3871</a><br/></div></div>

guilhermedcamacho commented on modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/view/table/DateTimeFDSTableSchemaField.java:
```diff
@@ -0,0 +1,27 @@
+
+	@Override
+	public String getContentRenderer() {
+		return "datetime";
```
Hey @markocikos , we had to change it to "dateTime" as requested by Brian. He merged it some hours ago [here](https://github.com/brianchandotcom/liferay-portal/pull/111367), please do a rebase when synced upstream. In the meantime @marcelabc or @gabrielwas can review the other parts of your code. Thanks a lot!

guilhermedcamacho commented on modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/frontend/data/set/view/table/ObjectDefinitionsTableFDSView.java:
```diff
@@ -0,0 +1,74 @@
+			fdsTableSchemaBuilder.addFDSTableSchemaField(
+				"dateModified", "modified-date");
+
+		dateFDSTableSchemaField.setContentRenderer("date");
```
Hey @markocikos , we had to change from "date" to "dateTime" here. Brian merged it some hours ago in the same [PR](https://github.com/brianchandotcom/liferay-portal/pull/111367) I've mentioned in the other comment, please do a rebase when synced upstream. In the meantime @marcelabc or @gabrielwas can review the other parts of your code. Thanks a lot!

markocikos commented:
@guilhermedcamacho Rebased, fixed a conflict, added requested changes in https://github.com/liferay-objects/liferay-portal/pull/298/commits/39f075508c6a4ca355cd87212a3da86aad4e6134, retested usages, force-pushed.

guilhermedcamacho commented:
ci:test:relevant

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/425/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/425/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 10 out of 10 jobs passed</h3><h3>:heavy_check_mark: ci:test:relevant - 22 out of 26 jobs passed in 1 hour 54 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/5d14220eeb2f5872e16fad989788951d478ba848">5d14220eeb2f5872e16fad989788951d478ba848</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/5d14220eeb2f5872e16fad989788951d478ba848">5d14220eeb2f5872e16fad989788951d478ba848</a><br/>Jenkins Build URL: <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)//2725">Acceptance Upstream DXP (master) #2725</a></p><details><summary><strong>ci:test:stable - 10 out of 10 jobs PASSED</strong></summary><details><summary><strong>10 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/18857/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/482075/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/368099/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/482073/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23606/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/3276/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23049/">test-portal-acceptance-pullrequest-batch(master)/portal-license-smoke-jdk8/0</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/451712/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/18858/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23609/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 21 out of 26 jobs PASSED</strong></summary><div><h4>5 Failed Jobs:</h4><ul><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/425/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23600/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8/0</strong></strike></a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/9318/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</strong></strike></a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23619/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8/0</strong></strike></a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/482077/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</strong></strike></a></li></ul></div><details><summary><strong>21 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/368097/">test-portal-acceptance-pullrequest-batch(master)/empty-osgi-core-dir-jdk8/0</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/20669/">test-portal-acceptance-pullrequest-batch(master)/empty-temp-dir-jdk8/0</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/20666/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/18857/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/482074/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/482075/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/424956/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8/0</a></li><li><a href="https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/368099/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8/0</a></li><li><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/482073/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8_stable/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/20371/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8/0</a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/20306/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8/0</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23606/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8_stable/0</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/451710/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8/0</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/479256/">test-portal-acceptance-pullrequest-batch(master)/patching-tool-jdk8/0</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/3276/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8/0</a></li><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23049/">test-portal-acceptance-pullrequest-batch(master)/portal-license-smoke-jdk8/0</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/451712/">test-portal-acceptance-pullrequest-batch(master)/rest-builder-jdk8/0</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/20663/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8/0</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/18858/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8/0</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/451709/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8/0</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23609/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8_stable/0</a></li></ul></details></details><h5>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/425/jenkins-report.html">here</a>.</h5><hr/><h4>This pull contains no unique failures.</h4><hr/><details><summary><strong>Failures in common with <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">acceptance upstream results</a> at 5d14220eeb2f5872e16fad989788951d478ba848:</strong></summary><ol><li><div><h5><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23600/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8/0</a></h5><div><h6>Job Results:</h6><p>6 Tests Passed.<br/>2 Tests Failed.</p></div><ol><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/23600//consoleText">AXIS_VARIABLE=0 #23600</a><ol><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/23600//testReport/com.liferay.portal.modules/ModulesStructureTest/testScanBuildScripts">com.liferay.portal.modules.ModulesStructureTest.testScanBuildScripts</a><pre><code>junit.framework.AssertionFailedError: Forbidden modules/test/poshi/poshi-standalone/gradle.properties
	at com.liferay.portal.modules.ModulesStructureTest$2.preVisitDirectory(ModulesStructureTest.java:229)
	at com.liferay.portal.modules.ModulesStructureTest$2.preVisitDirectory(ModulesStructureTest.java:168)
	at java.nio.file.Files.walkFileTree(Files.java:2677)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at com.liferay.portal.modules.ModulesStructureTest.testScanBuildScripts(ModulesStructureTest.java:166)
</code></pre></div></li><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/23600//testReport/com.liferay.portal.modules/ModulesStructureTest/testScanGradleFiles">com.liferay.portal.modules.ModulesStructureTest.testScanGradleFiles</a><pre><code>junit.framework.AssertionFailedError: Artifact dependency {classpath group: "com.liferay", name: "com.liferay.gradle.plugins.source.formatter", version: "latest.release"} in modules/test/poshi/poshi-standalone/build.gradle not permitted on master branch. Use a project dependency instead.
	at com.liferay.portal.modules.ModulesStructureTest._testGradleFile(ModulesStructureTest.java:1448)
	at com.liferay.portal.modules.ModulesStructureTest.access$800(ModulesStructureTest.java:68)
	at com.liferay.portal.modules.ModulesStructureTest$4.visitFile(ModulesStructureTest.java:398)
	at com.liferay.portal.modules.ModulesStructureTest$4.visitFile(ModulesStructureTest.java:372)
	at java.nio.file.Files.walkFileTree(Files.java:2670)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at com.liferay.portal.modules.ModulesStructureTest.testScanGradleFiles(ModulesStructureTest.java:370)
</code></pre></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/9318/">test-portal-acceptance-pullrequest-batch(master)/functional-tomcat90-mysql57-jdk8/0</a></h5><div><h6>Job Results:</h6><p>68 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/9318//consoleText">AXIS_VARIABLE=0 #9318</a><ol><li><div><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/9318//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_CreateObject_CanAddRelationOnRelationshipField_">LocalFile.CreateObject#CanAddRelationOnRelationshipField</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-12/test-1-35/test-portal-acceptance-pullrequest(master)/425/functional-tomcat90-mysql57-jdk8/0/0/LocalFile.CreateObject_CanAddRelationOnRelationshipField/index.html.gz">Poshi Report</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-12/test-1-35/test-portal-acceptance-pullrequest(master)/425/functional-tomcat90-mysql57-jdk8/0/0/LocalFile.CreateObject_CanAddRelationOnRelationshipField/summary.html.gz">Poshi Summary</a> - <a href="https://testray.liferay.com/reports/production/logs/2021-12/test-1-35/test-portal-acceptance-pullrequest(master)/425/functional-tomcat90-mysql57-jdk8/0/0/jenkins-console.txt.gz">Console Output</a></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/23619/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8/0</a></h5><div><h6>Job Results:</h6><p>3 Tests Passed.<br/>2 Tests Failed.</p></div><ol><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/23619//consoleText">AXIS_VARIABLE=0 #23619</a><ol><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/23619//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanOSGiLog">com.liferay.portal.log.assertor.PortalLogAssertorTest.testScanOSGiLog</a><pre><code>junit.framework.AssertionFailedError: 
Portal log assert failure, OSGi log found: /opt/dev/projects/github/liferay-portal/bundles/osgi/state/1640112061600.log:
!SESSION 2021-12-21 18:41:17.707 -----------------------------------------------
eclipse.buildId=unknown
java.version=1.8.0_121
java.vendor=Oracle Corporation
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_US

!ENTRY com.liferay.site.initializer.extender 4 0 2021-12-21 18:41:17.707
!MESSAGE FrameworkEvent ERROR
!STACK 0
org.osgi.framework.BundleException: Could not resolve module: com.liferay.site.initializer.extender [1034]
  Unresolved requirement: Import-Package: com.liferay.commerce.account.util; version="1.3.0"

	at org.eclipse.osgi.container.Module.start(Module.java:444)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1682)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1662)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1624)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1555)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)


	at com.life...</code></pre></div></li><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0/23619//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">com.liferay.portal.log.assertor.PortalLogAssertorTest.testScanXMLLog</a><pre><code>junit.framework.AssertionFailedError: 
FrameworkEvent ERROR 
org.osgi.framework.BundleException: Could not resolve module: com.liferay.site.initializer.extender [1034]
  Unresolved requirement: Import-Package: com.liferay.commerce.account.util; version="1.3.0"

	at org.eclipse.osgi.container.Module.start(Module.java:444)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1682)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1662)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1624)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1555)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)

	at com.liferay.portal.log.assertor.PortalLogAssertorTest.scanXMLLogFile(PortalLogAssertorTest.java:171)
	at com.liferay.portal.log.assertor.PortalLogAssertorTest$1.visitFile(PortalLogAssertorTest.java:98)
	at com.liferay.portal.log.assertor.PortalLogAssertorTest$1.visitFile(PortalLogAssertorTest.java:88)
	at java.nio.file.Files.walkFileTree(Files.java:2670)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at com.liferay.portal.log.asse...</code></pre></div></li></ol></div></li></ol></div></li><li><div><h5><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/482077/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8/0</a></h5><div><h6>Job Results:</h6><p>759 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1/482077//consoleText">AXIS_VARIABLE=1 #482077</a><ol><li><div><a href="https://test-1-16.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=1/482077//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">com.liferay.portal.log.assertor.PortalLogAssertorTest.testScanXMLLog</a><pre><code>junit.framework.AssertionFailedError: 
Unable to connect to a valid mail server. Please make sure one is properly configured: Couldn't connect to host, port: localhost, 25; timeout -1
	at com.liferay.portal.log.assertor.PortalLogAssertorTest.scanXMLLogFile(PortalLogAssertorTest.java:171)
	at com.liferay.portal.log.assertor.PortalLogAssertorTest$1.visitFile(PortalLogAssertorTest.java:98)
	at com.liferay.portal.log.assertor.PortalLogAssertorTest$1.visitFile(PortalLogAssertorTest.java:88)
	at java.nio.file.Files.walkFileTree(Files.java:2670)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at com.liferay.portal.log.assertor.PortalLogAssertorTest.testScanXMLLog(PortalLogAssertorTest.java:86)
</code></pre></div></li></ol></div></li></ol></div></li></ol></details></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/425/">test-portal-acceptance-pullrequest(master)#425</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/425/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-objects/liferay-portal/pull/298">liferay-objects#298</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1593185907">[master] ci:test:relevant - markocikos &gt; liferay-objects - PR#298 - 2021-12-21[08:55:16]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-35.liferay.com/job/publish-testray-report/1239/">publish-testray-report#1239</a><br/></div></div>

gabrielwas commented:
Just started reviewing :)

gabrielwas commented:
Rebased and resent in https://github.com/liferay-objects/liferay-portal/pull/321
